python/ray/_private/runtime_env/container.py

```diff
@@ -46,19 +46,28 @@ async def setup(
             "--userns=keep-id",
         ]
 
-        # The RAY_RAYLET_PID and RAY_JOB_ID environment variables are
-        # needed for the default worker.
-        container_command.append("--env")
-        container_command.append("RAY_RAYLET_PID=" + os.getenv("RAY_RAYLET_PID"))
+        # Environment variables to set in container
+        env_vars = dict()
+
+        # Propagate all host environment variables that have the prefix "RAY_"
+        # This should include RAY_RAYLET_PID
+        for env_var_name, env_var_value in os.environ.items():
+            if env_var_name.startswith("RAY_"):
+                env_vars[env_var_name] = env_var_value
+
+        # Support for runtime_env['env_vars']
+        env_vars.update(context.env_vars)
+
+        # Set environment variables
+        for env_var_name, env_var_value in env_vars.items():
```

Comment:
Is this the same behavior as non-container runtime env? Don't we only need to update env_vars specified from runtime env?

Reply:
You can ignore this. I misread code